templates/workloads/partials/cf/envvars.yml

```diff
@@ -19,6 +19,10 @@ Environment:
 - Name: COPILOT_MOUNT_POINTS
   Value: '{{jsonMountPoints .Storage.MountPoints}}'
 {{- end}}{{end}}
+{{- if .Publish}}{{- if .Publish.Topics}}
+- Name: COPILOT_PUBLISHERS
```

Comment:
Just a note here. In future when we are adding doc for this feature, we might want to include more detail on the format of this new env var.

Reply:
Should we update the name of this env var to `COPILOT_SNS_TOPIC_ARNS`?  
The publisher is the service itself, and we are publishing messages **to** sns topics or an eventbridge event bus.

Reply:
This JSON is of the format {topicName: [allAllowedWorkers]}, so I don't think ARNS would be a good name. How about COPILOT_SNS_TOPICS?

Reply:
oh interesting how will a customer use this environment variable?

My understanding is that any service with a `publish` field in their manifest will get this environment variable, and they'll be able to do something like this:
```go
var topics map[string]string
json.Unmarshal(os.Getenv('COPILOT_SNS_TOPIC_ARNS'), topics)

sns.Publish({
  Message: "created a new inventory item",
  TargetARN: topics["inventory"],
})
sns.Publish({
  Message: "created a new product item",
  TargetARN: topics["product"],
})
```
Basically the goal is to give a way for the service that's publishing messages to get the target arns, does that make sense? or am I misunderstanding something

Reply:
This variable was intended as a way to help populate worker init with topics available to that particular worker and to help customers know which topics are tied to which workers. Maybe I can take a middle ground here and either make the variable have something like {"topic name":{"topicArn":"arn", "workers":[]}}, or to have two separate variables, one for topic arns and one for topic workers?

Reply:
I think for starters we can just use a list of all SNS topics in the environment, as customers may have created a topic but not updated the "allowed_workers" in the publisher when they run `svc init` for their worker. Will CloudFormation fail on creation in this circumstance if the topic policy doesn't allow the queue to create a subscription?

Reply:
Added the code necessary to give a SNS Topic json of arns to envvars and added tests to ensure it works.